primefaces/src/main/java/org/primefaces/model/JpaLazyDataModel.java

```diff
@@ -235,9 +236,13 @@ protected Predicate createPredicate(FilterMeta filter, Field filterField,
                         ? cb.notEqual(fieldExpression, filterValueAsCollection.get().iterator().next())
                         : fieldExpression.in(filterValueAsCollection.get()).not();
             case BETWEEN:
-                throw new UnsupportedOperationException("MatchMode.BETWEEN currently not supported!");
+                Iterator<Object> iter1 = filterValueAsCollection.get().iterator();
```

Comment:
Minor but I like iterBetween and iterIn rather than 1amd 2

Reply:
Addressed this.